images/hook/entrypoint.sh

```diff
@@ -5,7 +5,30 @@ set -e
 # taking it from the environment variables
 
 echo "--> Assuming changeset from the environment: $RIG_CHANGESET"
-if [ $1 = "update" ]; then
+if [ $1 = "install" ]; then
+    echo "Checking: $RIG_CHANGESET"
+    if rig status $RIG_CHANGESET --retry-attempts=1 --retry-period=1s; then exit 0; fi
+
+    echo "Starting installation, changeset: $RIG_CHANGESET"
+    # deleting in case it has been already attempted
+    rig cs delete --force -c cs/$RIG_CHANGESET
```

Comment:
This is going to remove all changes made so far, so the restart of the hook (eg after some resources have already been added/updated) will use the updated resources instead of the original which is not what we want here.

Reply:
The install hook will be restarted automatically (due to the configured policy of `OnFailure`), so the next time the hook restarts and provided that it has failed after introduced new resources, this will remove the changeset that was previously started and use the new resources as original values if I understand correctly.

Reply:
afaik rigging will remove it's own changes (thus resetting the configmap as it was created via `--config`) and then restart from there.

I don't see how this could fail... maybe I'm missing something.

Reply:
The command `rig cs delete` simply deletes the changset CRD - it does not revert any changes made so far (it's what `rig revert` is for). For install hook I'd even tend towards using `kubectl apply/create` directly as `rig` does not bring any benefits for an install hook especially with restarts after a failure.